src/sort/sort.rs

```diff
@@ -125,11 +215,28 @@ With no FILE, or when FILE is -, read standard input.", NAME, VERSION);
         files.push("-".to_owned());
     }
 
+    settings.compare_fns.push(match settings.mode {
+        SortMode::Numeric => numeric_compare,
+        SortMode::HumanNumeric => human_numeric_size_compare,
+        SortMode::Month => month_compare,
+        SortMode::Version => version_compare,
+        SortMode::Default => String::cmp
+    });
+
+    if !settings.stable {
+        match settings.mode {
+            SortMode::Default => {}
+            _ => settings.compare_fns.push(String::cmp)
+        }
+    }
+
     exec(files, &settings)
 }
 
 fn exec(files: Vec<String>, settings: &Settings) -> i32 {
     let mut lines = Vec::new();
+    let mut file_merger = FileMerger::new(&settings);
```

Comment:
I ended up rewriting --check, so that it could be dragged out into its own function (see last commit in newly pushed version of branch).